pkg/operator/resources/validations.go

```diff
@@ -241,3 +258,52 @@ func getValidationK8sResources() ([]istioclientnetworking.VirtualService, kresou
 
 	return virtualServices, maxMem, err
 }
+
+// InclusiveFilterAPIsByKind includes only provided Kinds
+func InclusiveFilterAPIsByKind(apis []userconfig.API, kindsToInclude ...userconfig.Kind) []userconfig.API {
+	kindsToIncludeSet := strset.New()
+	for _, kind := range kindsToInclude {
+		kindsToIncludeSet.Add(kind.String())
+	}
+	fileredAPIs := []userconfig.API{}
+	for _, api := range apis {
+		if kindsToIncludeSet.Has(api.Kind.String()) {
+			fileredAPIs = append(fileredAPIs, api)
+		}
+	}
+	return fileredAPIs
+}
+
+// checkIfAPIExists checks if referenced apis in trafficsplitter are either defined in yaml or already deployed
+func checkIfAPIExists(trafficSplitterAPIs []*userconfig.TrafficSplit, apis []userconfig.API) error {
+	deployedAPIs, err := GetAPIs()
```

Comment:
This is a fairly expensive operation (and will get even more expensive after batch apis are added) because it gets a lot more information like the number of replicas running, metrics, etc. Since we are trying to figure out if the api's exist, we can do a `config.K8s.ListVirtualServicesByLabel("apiKind", userconfig.SyncAPI.String())` to get a list of virtualServices and then extract the api names. I would run this query once outside the for loop in ValidateClusterAPIs and then pass in the deployedAPINames list to this function.